### PR TITLE
Add Spellchecker

### PR DESCRIPTION
**Pull request by zslade**
### Type of PR

- [ ] BUG
- [x] FEAT
- [ ] MAINT
- [x] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

Addresses: #1587

### Give a brief description for the solution you have provided

I have set up [pyspelling spellchecker](https://facelessuser.github.io/pyspelling/) to run on docs. Other packages have been considered (`codespell`, `symspellpy` and various plugins), but `pyspelling` offers the overall best choice in terms of flexibility (can add words, e.g. Splink, to a shared custom dictionary + many configuration options), accuracy and speed.

The [dictionary against which words are compared is from LibreOffice](https://github.com/LibreOffice/dictionaries/blob/master/en/en_GB.aff). Justification: it is based on established source dictionaries (e.g. Oxford, Cambridge, Collins dictionary, etc) straightforward to download, well-maintained, widely used by many reputable softwares. Just as good, if not better, than alternatives considered (e.g. ONLYOFFICE).

All `pyspelling` related files are stored in `scripts/pyspelling`. These consist of:
- `spellchecker.sh` bash script for installing `pyspelling` reqs and running spellchecker easily from the command line
- `pyspelling.yml` config file. The Aspell spellchecker is chosen (this is `pyspelling` default) as this allows markdown mode and seems to perform just as well as the alternative, Hunspell.
- `custom_dictionary.txt` additional words to include in dictionary

I will add instructions to the docs on how to run the spellchecker if PR approved (and update the bash script with an option to just check one file as this will probably be useful)

N.B. `pyspelling` is currently only configured to work on `.md` files but there is scope for spellchecking `.ipynb` notebooks with future changes to the `.yml` config.

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [x] Made changes based off the latest version of Splink
- [x] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: scripts/pyspelling/custom_dictionary.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Splink
+PyPI
+DuckDB
+deduplication
+Conda
+Quickstart
```
Comment (ThomasHepworth) [inline scripts/pyspelling/custom_dictionary.txt:6]: Are these words case sensitive? I imagine not

**Review by ThomasHepworth — APPROVED** [review]
This works amazingly, thank you!

Just an FYI - I noticed the examples below which we can resolve in a follow-up PR.
Screenshot 2023-09-11 at 15 46 44

**Comment by ThomasHepworth** [comment]
If you merge master into this, the linting issues should go away.

File: scripts/pyspelling/custom_dictionary.txt (hunk 1 above)
Comment (zslade) [reply scripts/pyspelling/custom_dictionary.txt:6]: Yes they are case sensitive.

In the case of 'Splink' this is useful for picking up lowercase versions in the text, but this does mean that correct `splink`s in code lines will also be flagged as errors. Something that can be addressed in a follow-up PR me thinks

**Comment by zslade** [comment]
> This is works amazingly, thank you!
> 
> Just an FYI - I noticed the examples below which we can resolve in a follow-up PR. Screenshot 2023-09-11 at 15 46 44

Agreed. I've also spotted quite a few instances of this and other similar things

**Pull request merged**